### PR TITLE
Itinerary Narrative Semantic HTML

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
- Changes headers across the itinerary narrative and results views to `h1`
- Changes results to `ul`
- Adds headers and groupings to trip tools

### CODE REVIEW

File: lib/components/narrative/narrative-itineraries.js [hunk 1]
```diff
@@ -404,30 +406,37 @@ class NarrativeItineraries extends Component {
...
-                      this._renderItineraryRow(itin, true)
-                    )
-                  )}
-              </S.SingleModeRowContainer>
+              {groupItineraries && !itineraryIsExpanded && (
+                <li style={{ padding: '0 1em' }}>
+                  <h2>
+                    {intl.formatMessage({ id: 'otpUi.LocationField.other' })}
+                  </h2>
+                  <S.SingleModeRowContainer aria-label="Non-Transit Options">
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:428]: aria-label is redundant with heading above.

File: lib/components/narrative/narrative-itineraries.js [hunk 2]
```diff
@@ -404,30 +406,37 @@ class NarrativeItineraries extends Component {
               {groupItineraries && !itineraryIsExpanded
                 ? Object.keys(groupedMergedItineraries.multi).map((mode) => {
                     return (
-                      <div key={mode} style={{ padding: '0 1em' }}>
+                      <li key={mode} style={{ padding: '0 1em' }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:409]: I don't think the list is needed here, as headings already follow (2 instances).

File: lib/components/narrative/narrative-itineraries.js [hunk 3]
```diff
@@ -386,12 +388,12 @@ class NarrativeItineraries extends Component {
           showingErrors={showingErrors}
           sort={sort}
         />
-        <div
-          // FIXME: Change to a ul with li children?
-          className="list"
+        <ul
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:391]: I think this no longer needs to be a list (there are lists already under each mode heading).

File: lib/components/narrative/metro/metro-itinerary.tsx [hunk 4]
```diff
@@ -358,13 +358,10 @@ class MetroItinerary extends NarrativeItinerary {
 
     // Use first leg's agency as a fallback
     return (
-      <div
+      <li
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/metro/metro-itinerary.tsx:361]: When viewing one itinerary after a search, the viewed itinerary is still presented as a list with a single item. Instead, there should be no list. Therefore, `<li>` should move out of this component (retain `<div>` for now).

File: lib/components/narrative/narrative-itineraries.js [hunk 5]
```diff
@@ -404,30 +406,37 @@ class NarrativeItineraries extends Component {
...
-                      <div key={mode} style={{ padding: '0 1em' }}>
+                      <li key={mode} style={{ padding: '0 1em' }}>
                         <h2>{mode}</h2>
-                        {groupedMergedItineraries.multi[mode].map((itin) =>
-                          this._renderItineraryRow(itin)
-                        )}
-                      </div>
+                        <ul style={{ listStyle: 'none', padding: '0' }}>
+                          {groupedMergedItineraries.multi[mode].map((itin) =>
+                            this._renderItineraryRow(itin)
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:413]: Surround with `<li>` (assuming moving `<li>` out of `MetroItinerary`).

File: lib/components/narrative/narrative-itineraries.js [hunk 6]
```diff
@@ -404,30 +406,37 @@ class NarrativeItineraries extends Component {
...
-                {groupItineraries &&
-                  !itineraryIsExpanded &&
-                  Object.keys(groupedMergedItineraries.single).map((mode) =>
-                    groupedMergedItineraries.single[mode].map((itin) =>
-                      this._renderItineraryRow(itin, true)
-                    )
-                  )}
-              </S.SingleModeRowContainer>
+              {groupItineraries && !itineraryIsExpanded && (
+                <li style={{ padding: '0 1em' }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:424]: This list item is not needed if removing the parent `<ul>`.

File: lib/components/narrative/narrative-itineraries.js [hunk 7]
```diff
@@ -404,30 +406,37 @@ class NarrativeItineraries extends Component {
...
+                  <h2>
+                    {intl.formatMessage({ id: 'otpUi.LocationField.other' })}
+                  </h2>
+                  <S.SingleModeRowContainer aria-label="Non-Transit Options">
+                    {Object.keys(groupedMergedItineraries.single).map((mode) =>
+                      groupedMergedItineraries.single[mode].map((itin) =>
+                        this._renderItineraryRow(itin, true)
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:431]: Surround with `<li>` (assuming moving `<li>` out of `MetroItinerary`).

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Move `<li>` out of `MetroItinerary` and remove the list when rendering mode headings. Looks good otherwise.

File: lib/components/narrative/narrative-itineraries.js (hunk 3 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries.js:391]: Unfortunately I'm realizing the issue with turning this one back into a div is that in the batch-itinerary view, the results do not show up as a list

Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries.js:391]: Which probably means if this stays as a <ul>, the sub-lists of modes in the metro-itinerary should probably also remain?

**Review by miles-grant-ibigroup — APPROVED** [review]
One small nitpick but with that resolved this is good to go! Thanks for doing this finnicky work

File: lib/components/narrative/narrative-itineraries-header.tsx [hunk 8]
```diff
@@ -84,6 +88,9 @@ export default function NarrativeItinerariesHeader({
               <FormattedMessage id="components.NarrativeItinerariesHeader.viewAll" />
             </IconWithText>
           </button>
+          <h1 style={{ display: 'none' }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries-header.tsx:91]: For use with screen readers only, use CSS `visibility: hidden` and set a zero-width or zero-height (see other comment regarding `display: none`).

File: lib/components/narrative/narrative-itineraries.js [hunk 9]
```diff
@@ -256,6 +260,10 @@ class NarrativeItineraries extends Component {
 
     if (!activeSearch) return null
 
+    // render lists become divs if itinerary is expanded, to avoid rendering a list with list item
+    const ListContainer = itineraryIsExpanded ? 'div' : 'ul'
+    const ListItem = itineraryIsExpanded ? 'div' : 'li'
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries.js:265]: Not needed per https://github.com/opentripplanner/otp-react-redux/pull/744#discussion_r1063525664.

File: lib/components/narrative/narrative.css [hunk 10]
```diff
@@ -44,6 +44,20 @@
 }
 
 /* TRIP TOOLS ROW */
+
+.otp .trip-tools-container {
+  margin: 15px 5px;
+  padding: 10px 16px;
+  background-color: #eee;
+  border-radius: 6px;
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative.css:52]: Sort attributes.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
A few more touches needed, but we're almost there!

File: lib/components/narrative/narrative-itineraries.js [hunk 11]
```diff
@@ -197,6 +197,7 @@ class NarrativeItineraries extends Component {
       visibleItinerary
     } = this.props
     const { ItineraryBody, LegIcon } = this.context
+    const ListItem = itineraryIsExpanded ? 'div' : 'li'
```
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries.js:200]: Have to give some credit to @miles-grant-ibigroup for that one 😁

File: lib/components/narrative/narrative-itineraries.js (hunk 9 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries.js:265]: I think the same issue we pointed out in the above thread still applies - removing the outer <ul> means the batch-itin doesn't render as a list. Let me know if I'm misunderstanding!

Comment (binh-dam-ibigroup) [reply lib/components/narrative/narrative-itineraries.js:265]: To preserve that, how about moving the `ListContainer` at line 399 to the block around line 429 (make it surround `itinerariesWithCo2.map(...)`.

Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries.js]: That works great, thanks so much Binh!

File: lib/components/narrative/narrative-itineraries-header.tsx [hunk 12]
```diff
@@ -138,6 +151,7 @@ export default function NarrativeItinerariesHeader({
               marginLeft: showHeaderText ? 'inherit' : 'auto'
             }}
           >
+            <h1 style={{ display: 'none' }}>{itinerariesFound}</h1>
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/narrative/narrative-itineraries-header.tsx]: I am sorry, [`visibility: hidden` also hides content from screen readers](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#accessibility_concerns), so just keep the zero-size for these items.

File: lib/components/narrative/narrative-itineraries-header.tsx (hunk 8 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/narrative-itineraries-header.tsx]: Sorry, don't render this `<h1>` at all. It was not going to be shown to screen readers (see https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#accessibility_concerns). Rather, set a zero-size to the `<h1>` above this one as needed.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
My apologies... I gave you wrong advice regarding `visibility: hidden`.

File: lib/components/narrative/narrative-itineraries-header.tsx (hunk 8 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/narrative-itineraries-header.tsx]: Thank you so much for catching this!! I think removing this h1 leaves the itinerary narrative without a header, but I can fix both with `height: 0; width: 0; overflow: hidden;`

**Pull request merged**